### PR TITLE
Fix for Keycloak 22.0.1 unable to create user with long email address

### LINKED ISSUE #22825: Keycloak 22.0.1 unable to create user with long email address
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/api

### Describe the bug

With the latest nightly release, it is no more possible to create a user with email longer than 75 characters. Moreover in 22.0.1, the limit seems to be at 87 characters. 

Meanwhile in the old version 21.1.2, it is still possible to create a user with an email longer than 87 characters.

This happens both via REST API users and on the admin UI:
image

### Version

22.0.1

### Expected behavior

Still able to create user with more than 87 characters in the email like in 21.1.2. Not sure what was the limit before.

### Actual behavior

Unable to create user with more than 87 characters in the email.

### How to Reproduce?

- Run keycloak server
- Login and go to user section
- Try to create user with email address longer than 87 characters in 22.0.1 (or longer than 75 in nightly release)

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kaustubh-rh**
Closes #22825

### CODE REVIEW

**Comment by mposolda** [comment]
@kaustubh-rh Thanks! Could you please also cherry pick this commit https://github.com/keycloak/keycloak/commit/36dd9cb9375412f805a6af766d5c86eb8af90231 ? It is in latest Keycloak main and contains the changes needed for moving the stuff to 22.0.3 documentation.

So this PR would have 2 commits, which is fine.

**Comment by kaustubh-rh** [comment]
@mposolda  done

File: docs/documentation/upgrading/topics/keycloak/changes.adoc [hunk 1]
```diff
@@ -1,5 +1,16 @@
 == Migration Changes
 
+<<<<<<< HEAD
```
Comment (mposolda) [inline docs/documentation/upgrading/topics/keycloak/changes.adoc:3]: There are some issues in this file, which cause the documentation is impossible to build.

Could you please fix it and remove "conflict markers" from this file?

Also I suggest to manually build documentation on your laptop and checking it looks as expected before pushing it into the PR.

**Review by mposolda — CHANGES_REQUESTED** [review]
@kaustubh-rh Thanks. Looks good, but there are issues in the `changes.adoc` file. See inline comments.

File: docs/documentation/upgrading/topics/keycloak/changes.adoc (hunk 1 above)
Comment (kaustubh-rh) [reply docs/documentation/upgrading/topics/keycloak/changes.adoc]: Yes, I made the changes and then tried to build the documentation locally on my laptop and it's working fine.

**Review by mposolda — COMMENTED** [review]
@kaustubh-rh Could you please also add something like this:
```
=== Migrating to 22.0.4

include::changes-22_0_4.adoc[leveloffset=3]
```
to file `changes.adoc`, so the page for 22.0.4 is rendered as well?

**Comment by kaustubh-rh** [comment]
@mposolda added the same to changes.adoc. Thanks

**Review by mposolda — APPROVED** [review]
@kaustubh-rh Thanks! I hope to merge once tests are ok

**Pull request merged**